### DISCUSSION TITLE
Debuging the extension and uae-dap

### DISCUSSION THREAD

**Discussion by prb28** [prb28/vscode-amiga-assembly#294] [General]
Hi @grahambates,
What is your way to debug uae-dap with the extension ?
I've found this way : https://github.com/prb28/vscode-amiga-assembly/wiki/Building-Debugging-Extension but it's kind of complicated.

**Comment by grahambates** [comment]
The way I do it is to run the uae-dap project as a server, and then have the debug config connect to it.

Open the uae-dap project in vscode and run the 'Server' launch config":
<img width="413" alt="image" src="https://github.com/prb28/vscode-amiga-assembly/assets/1519709/923a0224-cbb3-4822-be45-1d0a4c525d54">

You'll see that it's waiting for a connection on the default port:
<img width="519" alt="image" src="https://github.com/prb28/vscode-amiga-assembly/assets/1519709/46fb77af-adf4-4d35-9b0c-4ea31fb196a7">

Then run the Amiga Assembly project (or just use the installed extension) and add the following to your debug config:
```json
"serverPort": 4711,
```
Now when you start debugging it will use the running server instead of starting a new instance, and you can set breakpoints etc in the uae-dap project.

This doesn't allow for testing issues with the webpack bundling etc, but it's good for testing functionality in the debug server itself.

I've also recently added a super lightweight debugger-only extension with the uae-dap project so you can run it without opening multiple projects, using the 'Extension + Server' launch config. This obviously doesn't help your use case of testing the integration with your extension though.

**Comment by prb28** [comment]
Thanks !
The option in the launch configuration was
`            "debugServer": 4711,`
After replacing the winuae executable from you master branch with the last released it worked.
Much easy to debug this way !
